### PR TITLE
Removing retry options from State APIs

### LINKED ISSUE #1861: Remove retry options from state apis.
Currently Dapr state apis provide option to specify retry policies. During review in Dapr maintainers meeting for state building block, it was decided to remove the retry policies for state as retries will be introduced in Dapr in a more cohesive fashion for all building blocks based on community feedback.
This issue to remove retry options from state apis before v1. This would need changes in the SDKs as well.

This issue tracks overall Dapr retry patterns: https://github.com/dapr/dapr/issues/1820

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

This PR removes  retry options from state apis before v1. 

## Issue reference

Closes #1861
Reference :  Overall Dapr retry patterns: #1820

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by youngbupark** [comment]
Can you remove grpc protobuf and regen it? or will you create pr separately?

File: pkg/grpc/api.go [hunk 1]
```diff
@@ -10,7 +10,6 @@ import (
 	"errors"
 	"fmt"
 	"strconv"
-	"time"
```
Comment (amanbha) [inline pkg/grpc/api.go:13]: Please update the protos as well and generated files in this PR.

Comment (shalabhms) [reply pkg/grpc/api.go:13]: updated the protos and added files in this PR.

**Comment by shalabhms** [comment]
> Can you remove grpc protobuf and regen it? or will you create pr separately?

Done now. It is part of this PR.

File: dapr/proto/common/v1/common.proto [hunk 2]
```diff
@@ -95,11 +94,11 @@ message StateItem {
   // The metadata which will be passed to state store component.
   map<string,string> metadata = 4;
 
-  // Options for concurrency, consistency, and retry_policy to save the state.
+  // Options for concurrency, and consistency to save the state.
```
Comment (amanbha) [inline dapr/proto/common/v1/common.proto:97]: nit, there is extra comma between concurrency & and

Comment (shalabhms) [reply dapr/proto/common/v1/common.proto]: fixed it now. thanks.

**Comment by shalabhms** [comment]
@youngbupark , I have removed grpc protobuf and regen it as part of this PR. Could you please take  a look ?

**Comment by yaron2** [comment]
lgtm

**Pull request merged**